ElGrecode/aaa-component-library
# Es lint prettier configuration

### Links
_Link to Jira ticket?_

### Description
_Detail with what was changed, why it was changed, and how it was changed_ 

FIRST OF ALL: IGNORE THE COMMIT PERTAINING TO Material UI.  I accidentally committed my changes to my shared component branch so when I did git reset head^ that got included in this PR too. It wont cause any build failure that is for sure. So kindly ignore it.
--**The problem**:    We have to have common setting to maintain code formatting across the app.
--**The Fix** Installing ESLint and Prettier


### How to test
The idea is every code commit it should run automatically for us so no need to test for it. See Additional notes for more info.

### Additional Notes
You can install extensions in your favourite IDE or code editor to highlight the eslint error and auto format the code accordingly to the rules you set

1. Eslint
2. Prettier

Settings

There are some settings you have to do for making these extensions workable.
{
  "files.autoSave": "onFocusChange",
  "editor.formatOnSave": true,
  "editor.formatOnType": true,
  "eslint.autoFixOnSave": true,
  "eslint.enable": true,
  "prettier.singleQuote": true,
}
